lab-brandon/MyArrayList.java

```diff
@@ -0,0 +1,64 @@
+import java.util.Arrays;
+
+public class MyArrayList {
+    private int DEFAULT_SIZE = 10;
+
+    private int[] data;
+    private int elements;
+
+    // Main constructor
+    public MyArrayList () {
+        this.data = new int[DEFAULT_SIZE];
+        this.elements = 0;
+    }
+
+    // Parameterized constructor
+    public MyArrayList (int customSize) {
+        this.data = new int[customSize];
+        this.elements = 0;
+    }
+
+    // adds a value to the end of the list
+    public int[] add (int val) {
+        this.data[this.elements] = val;
+        this.elements++;
+
+        return this.data;
+    }
+
```

Comment:
Your `addAtIndex` function as it is now will overwrite any existing values that might be at the given index; instructions state that it should "add the value at the index, shifting all other elements to make room."